base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -643,7 +650,24 @@ where B: BlockchainBackend
     pub fn prepare_block_merkle_roots(&self, template: NewBlockTemplate) -> Result<Block, ChainStorageError> {
         let NewBlockTemplate { header, mut body, .. } = template;
         body.sort();
-        let header = BlockHeader::from(header);
+        let mut header = BlockHeader::from(header);
```

Comment:
nit: We may as well call this function `prepare_new_block(NewBlocktemplate) -> Block`

Reply:
I agree, will rename. That is what the function did anyway. All we do now is we do an extra check before we set the timestamp. Before this, the set_timestamp happened in `let header = BlockHeader::from(header);`